### PR TITLE
HPCC-16517 Continue refactoring the roxie stats

### PR DESCRIPTION
**Pull request by ghalliday**

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review.

You probably want to merge these into master rather than 6.2.  I can re-target once happy and master is up to date.

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -1690,11 +1679,9 @@ class StrandProcessor : public CInterfaceOf<IEngineRowStream>
             if (inputStream)
                 inputStream->stop();
 
-            stats.reset(heapStatistics); // Heap stats are always gathered from scratch each time
-            if (rowAllocator)
-                rowAllocator->gatherStats(stats);
             parent.stop();
-            parent.mergeStrandStats(processed, totalCycles, stats);
+            //MORE: Move totalCycles (+processed?) to gatherStats()
```
Comment (richardkchapman) [inline roxie/ccd/ccdserver.cpp]: // MORE ..

**Review by richardkchapman — COMMENTED** [review]
There are a few places where //MORE comments suggest that there is more coming.

Other than that I didn't see anything that didn't look like a move in the right direction

**Comment by richardkchapman** [comment]
@ghalliday back to you

File: roxie/ccd/ccdserver.cpp [hunk 2]
```diff
@@ -532,10 +516,13 @@ class CRoxieServerActivityFactoryBase : public CActivityFactory, implements IRox
         CriticalBlock b(statsCrit);
         if (started)
             putStatsValue(&node, "_roxieStarted", "sum", started);
-        if (totalCycles)
-            putStatsValue(&node, "totalTime", "sum", (unsigned) (cycle_to_nanosec(totalCycles)/1000));
-        if (localCycles)
-            putStatsValue(&node, "localTime", "sum", (unsigned) (cycle_to_nanosec(localCycles)/1000));
+        //MORE: The following is information is now duplicated
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: This provides backward compatibility for anyone who is looking at the stats values.  Is that required, or shall I just delete this code?

**Comment by ghalliday** [comment]
@richardkchapman 
First //MORE I have deleted the code.
Second and third I clarified the comment
Fourth I opened a separate jira

**Comment by richardkchapman** [comment]
@ghalliday Needs rebasing for a clean merge

**Comment by ghalliday** [comment]
@richardkchapman rebased.

**Pull request merged**